Assets/Scripts/AttackLogic.cs

```diff
@@ -0,0 +1,175 @@
+
+        // -----temp replacement for animation-----
+            _gameObjectRenderer = gameObject.GetComponentInChildren<MeshRenderer>();
+            _prevMat = _gameObjectRenderer.material;
+            _attackAnimationMaterial = new Material(Shader.Find("Standard"));
+            _attackAnimationMaterial.color = Color.yellow;
+        // ----------
+    }
+
+    /// <summary>
+    /// If it has a target, attacks it.
+    /// </summary>
+    private void Update()
+    {
+        if (_target != null)
+        {
+            Attack();
+        }
+    }
+
+    /// <summary>
+    /// Attacks the target. If the target is not in attacking range, chases it. If the target is near
+    /// performs a hit on it. If currently performing a hit, waits for finishing the hit and deals 
+    /// damage if hit was successful. Either resets afterwards or continues chasing / attacking target.
+    /// </summary>
+    private void Attack()
+    {
+        
+
+        float distanceToTarget = Vector3.Distance(_target.transform.position, transform.position);
+        if (distanceToTarget < attackRange && !PerformingHit)
+        {
+            // is in range
+            _movable.StopMoving();
+            // face target
+            if (_movable.FaceTarget(_target, true))
+            {
+                // faces target
+                // perform hit
+                PerformingHit = true;
+
+                // -----temp-----
+                    _prevMat = _gameObjectRenderer.material;
+                
+                    _gameObjectRenderer.material = _attackAnimationMaterial;
+                // ----------
+
+            }
+        }
+        else if (PerformingHit)
+        {
+            bool? success = PerformHit();
+            if (success != null)
+            {
+                // hit animation performed
+                if (success == true)
+                {
+                    // deal damage
+                    IDamageable damageable = (IDamageable)_target.GetComponent<IDamageable>();
+                    damageable.Hit(damage);
+                }
+
+                // -----temp-----
+                    //Material newMaterial = new Material(Shader.Find("Standard"));
+                    //newMaterial.color = Color.gray;
+                    _gameObjectRenderer.material = _prevMat;
+                // ----------
+                
+                // end hit
+                PerformingHit = false;
+                if (resetAfterHit)
+                {
+                    // reset aggro independent of (un-)successful hit
+                    _target = null;
+                }
+            }
+        }
+        else
+        {
+            // chase target
+            _movable.Move(_target.transform.position);
+        }
+    }
+
+    /// <summary>
+    /// Starts the attack.
+    /// </summary>
+    /// <param name="target">The target to be attacked</param>
+    public void StartAttack(GameObject target)
```

Comment:
```suggestion
    public void StartAttack(IDamageable target)
```
Make it as specific as possible

Reply:
as stated above, same thing

Reply:
Could you please elaborate on that? It's not clear at all to me what you mean by that/what you're referring to 😅

Reply:
Oh okay it also didn't show me that you commented on existing comments... what I'm talking about is the discussion of whether to use IDamageable as target or GameObject (see the only open discussion above)